### PR TITLE
Linting config

### PR DESCRIPTION
**Pull request by GuillaumeRx**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,29 +74,45 @@
...
+    "prettier": "2.1.1",
+    "pretty-quick": "3.0.0",
+    "style-loader": "1.2.1",
+    "ts-loader": "8.0.2",
     "tsconfig-paths-webpack-plugin": "3.3.0",
-    "typescript": "^4.0.2"
+    "typescript": "4.0.2"
   },
   "dependencies": {
+    "@ethersproject/abstract-provider": "^5.0.3",
```
Comment (FrederikBolding) [inline package.json:104]: Can we make this a specific version?

File: package.json [hunk 2]
```diff
@@ -71,29 +74,45 @@
...
     "@electron-forge/plugin-webpack": "6.0.0-beta.53",
-    "@marshallofsound/webpack-asset-relocator-loader": "^0.5.0",
+    "@marshallofsound/webpack-asset-relocator-loader": "0.5.0",
     "@types/react": "16.9.49",
     "@types/react-dom": "16.9.8",
-    "@typescript-eslint/eslint-plugin": "^3.9.0",
-    "@typescript-eslint/parser": "^3.9.0",
-    "css-loader": "^4.2.1",
+    "@types/ws": "7.2.6",
+    "@typescript-eslint/eslint-plugin": "^3.6.1",
```
Comment (FrederikBolding) [inline package.json:81]: Can we make this a specific version?

File: package.json [hunk 3]
```diff
@@ -71,29 +74,45 @@
...
-    "eslint": "^7.6.0",
-    "eslint-plugin-import": "^2.20.0",
-    "fork-ts-checker-webpack-plugin": "^5.0.14",
-    "node-loader": "^1.0.1",
-    "style-loader": "^1.2.1",
-    "ts-loader": "^8.0.2",
+    "eslint": "7.6.0",
+    "eslint-config-airbnb-typescript": "9.0.0",
+    "eslint-config-prettier": "6.11.0",
+    "eslint-plugin-import": "^2.22.0",
```
Comment (FrederikBolding) [inline package.json:88]: Can we make all of these a specific version?

File: src/bridge.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+type Unsubscribe = () => void;
+type Listener = (...args: any[]) => void;
+
+interface IpcBridge {
+  send(channel: string, data: any): void;
+  subscribe(channel: string, listener: Listener): Unsubscribe;
+}
+
+export const IpcBridge = (ipcRenderer) => ({
```
Comment (FrederikBolding) [inline src/bridge.ts:9]: Missing typing

File: src/bridge.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    const subscription = (_: any, ...args: any[]) => listener(...args);
+    ipcRenderer.on(channel, subscription);
+
+    return () => {
+      ipcRenderer.removeListener(channel, subscription);
+    };
+  },
+});
+
+export const ipcBridge = (window as Window).ipcBridge as IpcBridge;
```
Comment (FrederikBolding) [inline src/bridge.ts:23]: Need a custom window type

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (vilsbole) [reply package.json:81]: Done

File: package.json (hunk 3 above)
Comment (vilsbole) [reply package.json:88]: Done

File: src/bridge.ts (hunk 5 above)
Comment (vilsbole) [reply src/bridge.ts]: Done

File: src/bridge.ts (hunk 4 above)
Comment (vilsbole) [reply src/bridge.ts]: Done

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**